examples/vae_simple.py

```diff
@@ -0,0 +1,322 @@
+            l_output_list.append(l_output)
+            if W_dec_hid is None:
+                W_dec_hid = l_dec_hid.W
+                b_dec_hid = l_dec_hid.b
+                W_dec_mu = l_dec_mu.W
+                b_dec_mu = l_dec_mu.b
+                W_dec_ls = l_dec_logsigma.W
+                b_dec_ls = l_dec_logsigma.b
+        l_Z_list.append(l_Z)
+    l_output = nn.layers.ElemwiseSumLayer(l_output_list, coeffs=1./L, name='output')
+    return l_enc_mu, l_enc_logsigma, l_dec_mu_list, l_dec_logsigma_list, l_output_list, l_output
+
+# ############################## Main program ################################
+
+# helper function for log-likelihood expression
+def log_likelihood(tgt, mu, ls):
+    return T.sum(-(np.float32(0.5 * np.log(2 * np.pi)) + ls)
+            - 0.5 * T.sqr(tgt - mu) / T.exp(2 * ls))
+
+def main(L=2, z_dim=2, n_hid=256, num_epochs=20, binary=True):
+    print("Loading data...")
+    X_train, X_val, X_test = load_dataset()
+    X_train_tgt = X_train.reshape(X_train.shape[0],-1)
+    X_val_tgt = X_val.reshape(X_val.shape[0],-1)
+    X_test_tgt = X_test.reshape(X_test.shape[0],-1)
+    width, height = X_train.shape[2], X_train.shape[3]
+    input_var = T.tensor4('inputs')
+    target_var = T.dmatrix('targets')
+
+    # Create VAE model
+    print("Building model and compiling functions...")
+    print("L = {}, z_dim = {}, n_hid = {}, binary={}".format(L, z_dim, n_hid, binary))
+    x_dim = width * height
+    l_z_mu, l_z_ls, l_x_mu_list, l_x_ls_list, l_x_list, l_x = \
+           build_vae(input_var, L=L, binary=binary, z_dim=z_dim, n_hid=n_hid)
+
+    # If there are dropout layers etc these functions return masked or non-masked expressions
+    # depending on if they will be used for training or validation/test err calcs
+    z_mu = lambda b: nn.layers.get_output(l_z_mu, deterministic=b)
+    z_ls = lambda b: nn.layers.get_output(l_z_ls, deterministic=b)
+    x_mu = lambda b: [nn.layers.get_output(l_x_mu, deterministic=b) for l_x_mu in l_x_mu_list]
+    x_ls = lambda b: [nn.layers.get_output(l_x_ls, deterministic=b) for l_x_ls in l_x_ls_list]
+    x_list = lambda b: [nn.layers.get_output(l_x1, deterministic=b) for l_x1 in l_x_list]
+
+    # Loss expression has two parts as specified in [1]
+    # kl_div = KL divergence between p_theta(z) and p(z|x)
+    # - divergence between prior distr and approx posterior of z given x
+    # - or how likely we are to see this z when accounting for Gaussian prior
+    # logpxz = log p(x|z)
+    # - log-likelihood of x given z
+    # - in binary case logpxz = cross-entropy
+    # - in continuous case, is log-likelihood of seeing the target x under the
+    #   Gaussian distribution parameterized by dec_mu, sigma = exp(dec_logsigma)
+    kl_div = lambda b: 0.5 * T.sum(1 + 2*z_ls(b) - T.sqr(z_mu(b)) - T.exp(2 * z_ls(b)))
+    if binary:
+        logpxz = lambda b: T.sum([nn.objectives.binary_crossentropy(x, target_var).sum()
+            for x in x_list(b)]) * (-1./L)
```

Comment:
I'd use `sum(nn.objectives.binary_crossentropy(x, target_var).sum() for x in x_list(b))` here, i.e., without `T.` and without square brackets. Maybe `T.sum()` turns this into a tensor now and summarizes it, instead of just directly adding up the costs.